api/v1alpha1/driver_types.go

```diff
@@ -0,0 +1,271 @@
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	// Pod's tolerations list
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type PluginResourcesSpec struct {
+	Registrar *corev1.ResourceRequirements `json:"registerer,omitempty"`
+	Liveness  *corev1.ResourceRequirements `json:"liveness,omitempty"`
+	Plugin    *corev1.ResourceRequirements `json:"plugin,omitempty"`
+}
+
+type PluginSpec struct {
+	// Embedded common pods spec
+	PodCommonSpec `json:"inline"`
+
+	// Driver's plugin daemonset update strategy, supported values are OnDelete and RollingUpdate.
+	// Default value is RollingUpdate with MaxAvailabile set to 1
+	UpdateStrategy *appsv1.DaemonSetUpdateStrategy `json:"updateStrategy,omitempty"`
+
+	// Resource requirements for plugin's containers
+	Resources PluginResourcesSpec `json:"resources,omitempty"`
+
+	Volumes []corev1.Volume `json:"pluginVolumes,omitempty"`
+
+	// kubelet directory path, if kubelet configured to use other than /var/lib/kubelet path.
+	KubeletDirPath string `json:"kubeletDirPath"`
+
+	// Control the host mount of /etc/selinux for csi plugin pods. Defaults to false
+	EnableSeLinuxHostMount *bool `json:"EnableSeLinuxHostMount,omitempty"`
+
+	// To indicate the image pull policy to be applied to all the containers in the csi driver pods.
+	ImagePullPolicy string `json:"imagePullPolicy"`
+}
+
+type ProvisionerResourcesSpec struct {
+	Attacher      *corev1.ResourceRequirements `json:"attacher,omitempty"`
+	Snapshotter   *corev1.ResourceRequirements `json:"snapshotter,omitempty"`
+	Resizer       *corev1.ResourceRequirements `json:"resizer,omitempty"`
+	Provisioner   *corev1.ResourceRequirements `json:"provisioner,omitempty"`
+	OMapGenerator *corev1.ResourceRequirements `json:"omapGenerator,omitempty"`
+	Liveness      *corev1.ResourceRequirements `json:"liveness,omitempty"`
+	Plugin        *corev1.ResourceRequirements `json:"plugin,omitempty"`
+}
+
+type ProvisionerSpec struct {
+	// Embedded common pods spec
+	PodCommonSpec `json:"inline"`
+
+	// Set replicas for csi provisioner deployment. Defaults to 2
+	Replicas *int32 `json:"provisionerReplicas,omitempty"`
+
+	// Resource requirements for provisioner's containers
+	Resources ProvisionerResourcesSpec `json:"resources,omitempty"`
+}
+
+type LivenessSpec struct {
+	// Port to expose liveness metrics
+	MetricsPort int `json:"metricsPort,omitempty"`
+}
+
+type LeaderElectionSpec struct {
+	// Duration in seconds that non-leader candidates will wait to force acquire leadership.
+	// Default to 137 seconds.
+	LeaseDuration int `json:"leaseDuration,omitempty"`
+
+	// Deadline in seconds that the acting leader will retry refreshing leadership before giving up.
+	// Defaults to 107 seconds.
+	RenewDeadline int `json:"renewDeadline,omitempty"`
+
+	// Retry Period in seconds the LeaderElector clients should wait between tries of actions.
+	// Defaults to 26 seconds.
+	RetryPeriod int `json:"retryPeriod,omitempty"`
+}
+
+type CephFsClientType string
```

Comment:
`CephFsClientType` to `CephFSClientType`

Reply:
FS is an abbreviation for filesystem and as we agreed before we will treat abbreviations as words to prevent the issue of multiple consecutive abbreviations. 

With that code styling rule in mind, it should be `Fs` and not `FS`